Playground: Implement a splitscreen

abc013 opened the pull request:
**Description**

Feature proposal for a (resizable) splitscreen for the playground:
![image](https://github.com/mrdoob/three.js/assets/21260178/2fbcd9df-2a3f-4b46-9207-7adfffbd1e85)

Currently, there are two ways to display results of the node trees:
1. One by pressing the leftmost button in the menu for a fullscreen-preview,
2. or by "isolating" (double-clicking) a single node such that only it and the result is shown.

Something I found annoying during my experiments is that often times, I wanted to see the results when plugging various nodes around for experimentation. The first display option requires an additional click/hotkey (and a little do-after because of the node-pop-up animation when the editor opens back up). The second option only allows to show a single node. A splitscreen allows to see the results of plugging instantly.

I am open for suggestions and discussion :)

abc013 committed:
Implement a splitscreen for the three.js playground

sunag commented on playground/SplitscreenManager.js:
```diff
@@ -0,0 +1,89 @@
+        this.gutter = document.createElement( "div" );
+        this.gutter.className = "gutter";
+
+        document.getElementById( "splitview" ).appendChild( this.gutter );
```
Would it be possible to use variables instead of obtaining elements by ID? This way you could create the div using javascript instead of adding it to the html.

abc013 replied:
Done :)

sunag commented:
- There appears to be a problem when the window is resized.
- When I click on preview it doesn't seem to return to the previous state of the split screen.

abc013 commented:
- Fixed. I had to wrap the renderer DOM element into another div for this, let me know if there is a better way to do this.
- Fixed, it does now return to the original view as demanded.

abc013 commented:
(The DeepScan check fails on an issue that seems to be totally unrelated to this PR - or am I overlooking something?)

Mugen87 commented:
A rebase would solve the issue. But yes, it's unrelated to your changes.

sunag commented on playground/index.html:
```diff
@@ -86,6 +114,11 @@
+				const container = document.createElement( 'div' );
+				container.className = 'splitview';
+
+				document.body.appendChild( container );
```
We can use `this.domElement` instead of `document.body` here and in others appends? This would make the editor more flexible if we added it embed in editor for exemple. I think we can do the same thing with the classes, for example: `flow splitview` intead of just `splitview`.

abc013 replied:
`this.domElement` is `null` in the context of the script (which is directly embedded into the html). Thus, I am not sure how to implement that, can you give me a hint?

sunag commented:
It's looking really cool, I think it would be the last details I saw.

sunag commented on playground/NodeEditor.js:
```diff
@@ -297,7 +339,7 @@ export class NodeEditor extends THREE.EventDispatcher {
 		previewMenu.add( editorButton );
 
-		this.domElement.append( menu.dom );
+		document.body.appendChild(menu.dom);
```
We need to move the menus to `domElement` in order to maintain the consistency of the hierarchy. It shoud be something like:
- editor class="editor"
  - splitview
  - f-menu
  - flow

sunag replied:
If you can follow the name style could be better intead of `div`, could remove some redundant classes name too. Something like this:

![image](https://github.com/mrdoob/three.js/assets/502810/08b881b7-4aab-4a43-8d04-73d235ece9e4)

`node-editor` is `domElement` in this case

- node-editor
  - f-preview ( instead of splitview )
  - flow
  - f-menu

abc013 replied:
Updated.
![image](https://github.com/mrdoob/three.js/assets/21260178/134cbf23-2c7f-48bb-a276-2e690486f7af)
f-menu still lives in flow, because otherwise i would have to keep track of when the editor gets added in order to then add f-menu to the domElement parent.

is it fine like that?
